// @uniswap/universal-router/typechain/factories/ILooksRare__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Signer } from "ethers";
import { Provider } from "@ethersproject/providers";

import type { ILooksRare } from "../ILooksRare";

export class ILooksRare__factory {
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): ILooksRare {
    return new Contract(address, _abi, signerOrProvider) as ILooksRare;
  }
}

const _abi = [
  {
    inputs: [
      {
        components: [
          {
            internalType: "bool",
            name: "isOrderAsk",
            type: "bool",
          },
          {
            internalType: "address",
            name: "taker",
            type: "address",
          },
          {
            internalType: "uint256",
            name: "price",
            type: "uint256",
          },
          {
            internalType: "uint256",
            name: "tokenId",
            type: "uint256",
          },
          {
            internalType: "uint256",
            name: "minPercentageToAsk",
            type: "uint256",
          },
          {
            internalType: "bytes",
            name: "params",
            type: "bytes",
          },
        ],
        internalType: "struct TakerOrder",
        name: "takerBid",
        type: "tuple",
      },
      {
        components: [
          {
            internalType: "bool",
            name: "isOrderAsk",
            type: "bool",
          },
          {
            internalType: "address",
            name: "signer",
            type: "address",
          },
          {
            internalType: "address",
            name: "collection",
            type: "address",
          },
          {
            internalType: "uint256",
            name: "price",
            type: "uint256",
          },
          {
            internalType: "uint256",
            name: "tokenId",
            type: "uint256",
          },
          {
            internalType: "uint256",
            name: "amount",
            type: "uint256",
          },
          {
            internalType: "address",
            name: "strategy",
            type: "address",
          },
          {
            internalType: "address",
            name: "currency",
            type: "address",
          },
          {
            internalType: "uint256",
            name: "nonce",
            type: "uint256",
          },
          {
            internalType: "uint256",
            name: "startTime",
            type: "uint256",
          },
          {
            internalType: "uint256",
            name: "endTime",
            type: "uint256",
          },
          {
            internalType: "uint256",
            name: "minPercentageToAsk",
            type: "uint256",
          },
          {
            internalType: "bytes",
            name: "params",
            type: "bytes",
          },
          {
            internalType: "uint8",
            name: "v",
            type: "uint8",
          },
          {
            internalType: "bytes32",
            name: "r",
            type: "bytes32",
          },
          {
            internalType: "bytes32",
            name: "s",
            type: "bytes32",
          },
        ],
        internalType: "struct MakerOrder",
        name: "makerAsk",
        type: "tuple",
      },
    ],
    name: "matchAskWithTakerBidUsingETHAndWETH",
    outputs: [],
    stateMutability: "payable",
    type: "function",
  },
];
